turbopack/crates/turbopack-core/src/issue/mod.rs

```diff
@@ -512,6 +542,76 @@ impl IssueSource {
     }
 }
 
+async fn source_pos(
+    source: Vc<Box<dyn Source>>,
+    origin: Vc<FileSystemPath>,
+    start: SourcePos,
+    end: SourcePos,
+) -> Result<Option<(Vc<Box<dyn Source>>, SourcePos, SourcePos)>> {
+    let Some(generator) = Vc::try_resolve_sidecast::<Box<dyn GenerateSourceMap>>(source).await?
+    else {
+        return Ok(None);
+    };
+
+    let srcmap = generator.generate_source_map();
+    let Some(srcmap) = *srcmap.await? else {
+        return Ok(None);
+    };
+
+    // let Some(srcmap) = *convert_to_turbopack_source_map(srcmap, origin).await? else {
+    //     return Ok(None);
+    // };
+
+    let find = |line: usize, col: usize| async move {
+        let token = srcmap.lookup_token(line, col).await?;
+
+        match &*token {
+            crate::source_map::Token::Synthetic(t) => Ok::<_, anyhow::Error>((
+                t.guessed_original_file.clone(),
```

Comment:
generated_line and generated_column are not line numbers in guessed_original_file